cfgov/unprocessed/apps/owning-a-home/js/explore-rates/RateCheckerChart.js

```diff
@@ -0,0 +1,206 @@
+        }
+      } ],
+      series: [ {
+        name: 'Number of Lenders',
+        data: [ 1, 1, 1, 1, 1 ],
+        showInLegend: false,
+        dataLabels: {
+          enabled:   true,
+          useHTML:   true,
+          crop:      false,
+          overflow:  'none',
+          defer:     true,
+          color:     '#919395',
+          x:         2,
+          y:         2,
+          formatter: function() {
+            const point = this.point;
+            window.setTimeout( function() {
+              if ( point.y > 9 ) {
+                point.dataLabel.attr( {
+                  y: -32,
+                  x: point.plotX - 24
+                } );
+              }
+            } );
+            return '<div class="data-label"><span class="data-label_number">' + this.x + '</span><br>|</div>';
+          }
+        }
+      } ],
+      credits: {
+        text: ''
+      },
+      tooltip: {
+        useHTML: true,
+        formatter: function() {
+          if ( this.y === 1 ) {
+            return chartTooltipSingle( this );
+          }
+
+          return chartTooltipMultiple( this );
+        },
+        positioner: function( boxWidth, boxHeight, point ) {
+          let x, y;
+          if ( point.plotY < 0 ) {
+            x = point.plotX - 74;
+            y = this.chart.plotTop - 74;
+          } else {
+            x = point.plotX - 54;
+            y = point.plotY - 66;
+          }
+          return {
+            x: x,
+            y: y
+          };
+        }
+      }
+    } );
+  }
+
+  /**
+   * Render (or update) the _highCharts chart.
+   * @param  {Object} data Data processed from the API.
+   */
+  function render( data ) {
+    if ( isInitialized ) {
+      _highChart.update( {
+        xAxis: {
+          categories: data.labels
+        },
+        series: {
+          data: data.vals
+        }
+      } );
+      _containerDom.classList.remove( 'geolocating' );
+    } else {
+      _containerDom.classList.add( 'geolocating' );
+      _createHighcharts();
+      isInitialized = true;
+    }
+  }
+
+  /**
+   * Set the state of the slider.
+   * @param {number} state 0 = okay, 1 = warning state.
+   */
+  function setStatus( state ) {
+    if ( state === RateCheckerChart.STATUS_OKAY ) {
+      _chartDom.classList.remove( 'warning' );
+      _resultAlertDom.classList.add( 'u-hidden' );
+      _failAlertDom.classList.add( 'u-hidden' );
+    } else if ( state === RateCheckerChart.STATUS_WARNING ) {
+      _chartDom.classList.add( 'warning' );
+      _resultAlertDom.classList.remove( 'u-hidden' );
+    } else if ( state === RateCheckerChart.STATUS_ERROR ) {
+      _failAlertDom.classList.remove( 'u-hidden' );
+    } else {
+      throw new Error( `Status ${ state } set in chart is not supported!` );
+    }
+
+    _currentState = state;
+  }
+
+  /**
+   * Show an error alert.
+   * @param {number} errorType 1 = warning, 2 = error.
+   */
+  function stopLoadingShowError( errorType ) {
```

Comment:
How about naming this `throwLoadingError`

Reply:
Also, should the `loaded` class really be added if it's an error? Seems we need another function here to remove `loading` and maybe add an `error` class.

Reply:
I got rid of it and just made the error an optional parameter of `finishLoading`. This is actually how it was before.